app/controllers/users/registrations_controller.rb

```diff
@@ -1,5 +1,11 @@
 module Users
   class RegistrationsController < Devise::RegistrationsController
+
+    def edit
+      authorize current_user if their_own?
```

Comment:
This is what the `UserPolicy` responsibility is, so it should only be something like this:

```ruby
authorize current_user, :edit_password?
```

And then in the `UserPolicy`:

```ruby
class UserPolicy < BasePolicy
  ....
  def edit_password?
    user_themselves?
  end
  ....
end
```

Plus specs for the policy.